crates/compiler/gen_llvm/src/llvm/lowlevel.rs

```diff
@@ -2168,12 +2168,13 @@ fn build_dec_unary_op<'a, 'ctx>(
     use roc_module::low_level::LowLevel::*;
 
     match op {
-        NumSin => dec_unary_op(env, bitcode::DEC_SIN, arg),
-        NumCos => dec_unary_op(env, bitcode::DEC_COS, arg),
-        NumTan => dec_unary_op(env, bitcode::DEC_TAN, arg),
-        NumAsin => dec_unary_op(env, bitcode::DEC_ASIN, arg),
+        NumAbs => dec_unary_op(env, bitcode::DEC_ABS, arg),
```

Comment:
I feel like this code could better leverage Rust's type system in at least two ways:
1. Since all "lowlevel" operations are bunched up in the same enum, no pattern matching of this type can be explicitly exhaustive, as e.g. Dec deoesn't implement StrConcat. It would be better to have the `LowLevel` enum instead be something like:
```rust
pub enum LowLevel {
    Num(NumOp),
    Str(StrOp),
    ...
}

enum NumOp {
    Add,
    Mul,
    ...
}

...
```
2. More of a nitpick, but functions like `build_dec_unary_op` suggest that traits should be used instead - something like a `Num` trait that `Dec`, `Int` etc. implement.

Does that sound fair? Can I, in a separate PR, refactor the typing in lowlevel.rs accordingly?

Reply:
I think this is a pretty reasonable idea, but I think numerics would need to be split farther if you are trying to reach exhaustiveness. For example, some ops may run on integer but on float types.